core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala

```diff
@@ -346,9 +346,15 @@ class ModelInsightsTest extends FlatSpec with PassengerSparkFixtureTest {
             i.featureName shouldEqual o.featureName
             i.featureType shouldEqual o.featureType
             i.derivedFeatures.zip(o.derivedFeatures).foreach{ case (ii, io) => ii.corr shouldEqual io.corr }
-            RawFeatureFilterResultsComparison.compareSeqDistributions(i.distributions, o.distributions)
-            RawFeatureFilterResultsComparison.compareSeqExclusionReasons(i.exclusionReasons, o.exclusionReasons)
         }
+        RawFeatureFilterResultsComparison.compareSeqDistributions(
```

Comment:
**Note**: if there is some Double.NaN present here, while it may get correctly serialized and deserialized, this test will erroneously fail because it will not consider them equal

We can try to create some cases to correctly return True for NaN in this test, but I don't think this is a practical issue

Reply:
NaN value equality checks should have been handled by this implicit - https://github.com/salesforce/TransmogrifAI/blob/master/core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala#L57

did it not work for your test?

Reply:
Ah, didn't see those implicits before. Just tested it and not sure why they aren't being picked up by the equality test.

A test with
```scala
Some(Double.NaN) shouldEqual Some(Double.NaN)
```
will fail with error

`org.scalatest.exceptions.TestFailedException: Some(NaN) did not equal Some(NaN)`

But the logic inside the equality check is correct: a direct test of 

```scala
areEqual(Some(Double.NaN), Some(Double.NaN))
```
returns true